src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -91,7 +91,14 @@ class AppsResource @Inject() (
   def create(@Context req: HttpServletRequest, body: Array[Byte],
              @DefaultValue("false")@QueryParam("force") force: Boolean): Response = {
     val app = Json.parse(body).as[AppDefinition]
-    create(req, app, force)
+    service.getApp(app.id) match {
+      case None => create(req, app, force)
+      case _ => Response.status(409).entity(
+        Json.obj(
+          "message" -> "app with same id already exists"
```

Comment:
Yes, I will need to rebase with the other changes anyways, so I will just combine it with this change. Thanks!